File: Jt76.ExternalServices/WeatherServices/Models/WeatherData.cs
namespace Jt76.ExternalServices.WeatherServices.Models
{
	using System;
	using System.Collections.Generic;

	public class WeatherData
	{
		public string Description { get; set; }

		public DateTime Sunrise { get; set; }

		public DateTime Sunset { get; set; }

		public string City { get; set; } = "Unknown";

		public string Country { get; set; } = "USA";

		public ICollection<Forecast> WeatherForecasts { get; set; }
	}
}

File: Jt76.Ui/ClientApp/src/app/modules/weather/models/weather-data.ts
import { IWeatherForecast } from "./weather-forecast";

export interface IWeatherData {
  city: string;
  country: string;
  description: string;
  sunrise: Date;
  sunset: Date;
  weatherForecasts: Array<IWeatherForecast>;
}

export class WeatherData implements IWeatherData {
  city = "";
  country = "";
  description = "";
  sunrise = new Date();
  sunset = new Date();
  weatherForecasts = new Array<IWeatherForecast>();
}
